docs/developer/pull-requests.md

```diff
@@ -3,6 +3,142 @@ This document outlines best practices to follow when creating and reviewing PRs.
 
 ## As an author
 
+### Pull request process
+1. Create a GitHub PR from your fork
+1. Verify all checks are passing
+1. Get someone to review it
+1. Iterate your changes based on the feedback you receive. Don't rewrite the history (rebase or squash) with your new changes, keep them as individual commits. This facilitates the reviewers job.
+1. Once you haver al least one `lgtm` and are ready to merge, squash your commits into one.
```

Comment:
Imposing a 1 commit per PR requirement isn't good. I, for example, manually structure commits and put a lot of thought into them. That effort is wasted and even problematic when I want to go back and perhaps revert a particular change I made.

Reply:
These changes are not introducing that imposition though, we have prow a GitHub already configured that way. So unless I'm mistaken, even if you don't currently squash your commits manually, they will be automatically squashed before merging (I think I mention this a bit bellow). My main goal was to get the best commit history possible under the current constrains.

That said, if you want to reopen that debate (which includes merge commits, etc.), that's cool with me 🙂

Reply:
Resolved offline.